csrc/scheduler/registry.cpp

```diff
@@ -435,6 +439,78 @@ class SchedulerTopologyChecker {
 
     return false;
   }
+
+  // Checks if there's any gather-like ops that result in non-resolved
+  // broadcast domains and then get squeezed before reaching reduction
+  // TVs. The reduction scheduler uses reduction TVs as a scheduling
+  // reference, so that won't be able to schedule the broadcast ID if
+  // squeezed and its corresponding index-accessed producer ID, and
+  // any IDs that the producer ID depends on.
+  //
+  // This analysis has some similarity as DomainMap. Can be
+  // consolidated?
```

Comment:
It don't have to be this PR, but I do think at some point we would need to cleanup this and reuse the `DomainMap` from pointwise scheduler. Also, we should cache the `DomainMap`.

Reply:
I think all of these analyses should be replaced with the new `ComputeAtMap` as it should provide these traversal APIs.